buildroot/share/cmake/CMakeLists.txt

```diff
@@ -97,7 +97,7 @@ setup_motherboard(${PROJECT_NAME} ${CMAKE_CURRENT_LIST_DIR}/../../../Marlin)
 #====================================================================#
 
 file(GLOB SOURCES "../../../src/*.cpp")
```

Comment:
Shoudln't that one be `../../../Marlin` instead of `../../../src`, too?

Reply:
🤷 I haven't gotten far enough to check on that. And it seems like this `GLOB` value isn't going to be relevant to our new complex file layout.

Reply:
Maybe `../../../Marlin/**/*.cpp` then?

Reply:
If that style of path can apply in this context….

Reply:
Okay, so `file(GLOB_RECURSE SOURCES "../../../Marlin/*.cpp")` is what we want here.

See: https://cmake.org/cmake/help/latest/command/file.html#glob-recurse

Reply:
Good to know, thanks! Ultimately, it would be nice if these make files could filter sources according to enabled features, the way we currently do with PlatformIO.

Reply:
Unless some sort of configure script extracts the Configuration.h defines and feed them to the Makefile, I don't see how this could be done.